### PR TITLE
NullDAO constructor taking 'of'. 

### PR DESCRIPTION
**Pull request by jlhughes**

### CODE REVIEW

File: src/foam/dao/NullDAO.js [hunk 1]
```diff
@@ -22,6 +22,18 @@ foam.CLASS({
 
   documentation: 'A Null pattern (do-nothing) DAO implementation.',
 
+  properties: [
+    {
+      name: 'of',
+      factory: function() {
+        return this.cls_;
```
Comment (kgrgreer) [inline src/foam/dao/NullDAO.js]: Isn't this going to return NullDAO? Is that useful? Better to set value: foam.core.FObject?

Comment (jlhughes) [reply src/foam/dao/NullDAO.js]: Had to pick something.  Need to satisfy the getOf().getId() in PMInfoDAO. 
I'll update. 
What should the js be?  just   return foam.core.FObject;   ?
In java FObject is an interface.  Not sure what to use or call?

Comment (kgrgreer) [reply src/foam/dao/NullDAO.js]: Does AbstractFObject work?

**Comment by jlhughes** [comment]
Use Builder instead.

**Pull request closed without merging**